readthedocs/projects/models.py

```diff
@@ -463,6 +464,29 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         except Exception:
             log.exception('Error creating default branches')
 
+    def delete(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        from readthedocs.projects import tasks
+
+        # Remove local FS build artifacts on the web servers
+        broadcast(
+            type='app',
+            task=tasks.remove_dirs,
+            args=[(self.doc_path,)],
+        )
+
+        # Remove build artifacts from storage
+        storage_paths = []
+        for type_ in MEDIA_TYPES:
+            storage_paths.append(
+                '{}/{}'.format(
```

Comment:
I think it's better to use either `os.path.join` or `pathlib.Path` here.

Reply:
This is not a path on the local disk. This is a storage path which is different. It always uses forward slashes regardless of platform.

Reply:
If we want to go fancy, what we need here is a [`PurePosixPath`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePosixPath): it's platform independent and does not access the filesystem when manipulating paths.

Reply:
The Django storage API does not implement a posix standard either. I believe that using pathlib is not the correct course of action because these are not filesystem paths.